### ISSUE TITLE
Custom #load to load directly from web

### ISSUE THREAD

**Issue by marking** [scriptcs/scriptcs#1168]
Any chance something like this http://www.strathweb.com/2016/06/implementing-custom-load-behavior-in-roslyn-scripting/ can be incorporated? It allows for loading csx files over http, e.g. #load http://mysite.com/libraries/functions.csx.

**Comment by glennblock** [comment]
Absolutely. There is already a scriptcs gist module that lets you load from
a gist.

https://github.com/ryanrousseau/ScriptCs.Git

You could write something similar for general http.

Glenn

**Comment by marking** [comment]
That link is 404 for me. I'm assuming you meant https://github.com/ryanrousseau/ScriptCs.Gisit :smile: 

I'll take a look at implementing it in that manner. I was hoping to just get #load modified to support both file paths and http[s] paths, however.

**Comment by glennblock** [comment]
https://github.com/ryanrousseau/ScriptCs.Gist

That is also possible. You could create a derived LoadLineProcessor and
register it in a module.

https://github.com/scriptcs/scriptcs/blob/dev/src/ScriptCs.Core/LoadLineProcessor.cs

**Comment by glennblock** [comment]
Or if you want, you could send us a PR to the existing one and add support
for urls.

**Comment by AmadeusW** [comment]
> I was hoping to just get #load modified to support both file paths and http[s] paths, however.

I'd like that too, but given...

> #load is not a scriptcs invention, but rather a C# REPL standard. As a consequence of this, if you have Roslyn CTP installed and you use C# Interactive Window, you can copy paste you CSX file there and the #load directive will be correctly recognized.

via [wiki: Referencing scripts](https://github.com/scriptcs/scriptcs/wiki/Referencing-scripts)
also see [filipw's comment](https://github.com/scriptcs/scriptcs/issues/109#issuecomment-14864266)

...is it necessary to not break `#load` and instead introduce custom directives? I need this functionality and I'm curious if you would accept PR that introduces `#http` or `#loadhttp` directive?

**Comment by glennblock** [comment]
#load is a scriptcs invention WITHIN scriptcs. We don't use Roslyn's #load, we implement our own so that it works on Mono and Roslyn.

If you want to make #load support a url, you would need to modify the [LoadLineProcessor](https://github.com/scriptcs/scriptcs/blob/dev/src/ScriptCs.Core/LoadLineProcessor.cs). If you want to send us a PR with support for pulling down from URLs, go for it.

**Comment by glennblock** [comment]
Alternatively you could create a new directive, #loadurl. Either would work.
